cpp/src_prims/random/make_blobs.cuh

```diff
@@ -26,46 +28,120 @@
 namespace MLCommon {
 namespace Random {
 
+namespace {
+
+// generate the labels first and shuffle them instead of shuffling the dataset
+template <typename IdxT>
+void generate_labels(IdxT* labels, IdxT n_rows, IdxT n_clusters, bool shuffle,
```

Comment:
Now that I think more. If we just generated those a and b values, we'd mostly be doing the exact same thing as in `generate_labels`. Probably not a good way to test this code. Instead, how about the approach took as described below (ie revamped the make_blobs unit-test itself)?

Reply:
I agree, I was thinking about that later on too. I concluded at the moment that if we could sequentially generate the same shuffled labels as were generated parallelly (with seed), it would be deterministic. However, your test is much better!